ext/rotoscope/rotoscope.c

```diff
@@ -119,17 +120,34 @@ static rs_tracepoint_t extract_full_tracevals(rb_trace_arg_t *trace_arg,
 
 static bool in_fork(Rotoscope *config) { return config->pid != getpid(); }
 
+VALUE escape_csv_string(VALUE string) {
+  if (!memchr(RSTRING_PTR(string), '"', RSTRING_LEN(string))) {
+    return string;
+  }
+  return rb_funcall(string, id_gsub, 2, str_quote, str_escaped_quote);
+}
+
 static void log_trace_event(FILE *stream, rs_tracepoint_t *trace) {
-  fprintf(stream, RS_CSV_FORMAT "\n", RS_CSV_VALUES(trace));
+  VALUE escaped_method_name = escape_csv_string(trace->method_name);
+  fprintf(stream, RS_CSV_FORMAT "\n",
+          RS_CSV_VALUES(trace, escaped_method_name));
+  RB_GC_GUARD(escaped_method_name);
```

Comment:
What is this RB_GS_GUARD for ? Can't find doc on it

Reply:
It is to tell Ruby to not GC this value https://github.com/ruby/ruby/blob/96db72ce38b27799dd8e80ca00696e41234db6ba/doc/extension.rdoc#appendix-e-rb_gc_guard-to-protect-from-premature-gc